### PR TITLE
feat: adding ProviderType

### PR DESCRIPTION
**Pull request by 2fd**

### CODE REVIEW

**Review by cazala — APPROVED** [review]

File: src/dapps/provider-type.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  WALLET_CONNECT = 'wallet_connect'
+}
+
+/**
+ * @alpha
+ */
+export namespace ProviderType {
+  export const schema: JSONSchema<ProviderType> = {
+    type: 'string',
+    enum: Object.values(ProviderType)
```
Comment (menduz) [inline src/dapps/provider-type.ts:20]: Does this work? I've never added a Iterable to an enum, only arrays

Comment (2fd) [reply src/dapps/provider-type.ts:20]: you have one here: https://github.com/decentraland/common-schemas/blob/main/src/dapps/network.ts

Comment (2fd) [reply src/dapps/provider-type.ts:20]: https://github.com/decentraland/common-schemas/blob/main/src/dapps/network.ts

Comment (2fd) [reply src/dapps/provider-type.ts:20]: we can change it to:
```ts
  export const schema: JSONSchema<ProviderType> = {
    type: 'string',
    enum: [
      ProviderType.INJECTED,
      ProviderType.FORTMATIC,
      ProviderType.NETWORK,
      ProviderType.WALLET_CONNECT,
    ]
  ]
```

**Pull request merged**